### ISSUE TITLE
How to use with Laravel?

### ISSUE THREAD

**Issue by fabnoe** [Andreyco/Instagram-for-PHP#12]
How do I use the library in a Laravel Controller? I am not sure how to Inject the Class. Can you post an example?

Thanks,
Fabian

**Comment by Andreyco** [comment]
Hi there.

Simply require with `composer` 
```bash
composer require andreyco/instagram
```

and follow instructions here: https://github.com/Andreyco/Instagram-for-PHP#laravel

**Comment by Andreyco** [comment]
@fabnoe did you find it any helpful?

**Comment by PanMan** [comment]
For me the laravel setup doesn't work. I did get it to work by setting it up explicitly (eg the 'normal PHP') example. When setting up the Laravel version none of the settings (eg API key and secret) are applied. Might have to do that this project is still on Laravel 5.2

**Comment by Andreyco** [comment]
I will try to update to 5.3+ as soon as possible. Open to PR as well

**Comment by doug-proctor** [comment]
@fabnoe, @PanMan, the instructions given by @Andreyco [update: almost] work for me with Laravel 5.4. Here they are in a bit more detail:

1. Install the package by running the command `composer require andreyco/instagram`

2. Open **config/app.php** and register the service by adding the following entry below the "providers" array key: `'Andreyco\Instagram\Support\Laravel\ServiceProvider\Instagram',`

3. Also in **config/app.php**, add `'Instagram' => 'Andreyco\Instagram\Support\Laravel\Facade\Instagram',` below the "aliases" array key. 

4. The Instagram login URL should now be functional—add it to one of your views: `@php echo "<a href='" . Instagram::getLoginUrl() . "'>Login with Instagram</a>"; @endphp`.

UPDATE: I jumped the gun. Two more steps required:

5.  Run `php artisan vendor:publish` in your terminal. This will create the file **config/instagram.php** which will house your client id and secret. 

6. _Note: This is where a modification to the package is required; I have submitted a fix in PR #16 . Open **vendor/andreyco/instagram/src/Support/Laravel/ServiceProvider/instagram.php** and in the `register()` method, replace the returned array contents with:

```
'apiKey'      => $app['config']->get('instagram.clientId'),
'apiSecret'   => $app['config']->get('instagram.clientSecret'),
'apiCallback' => $app['config']->get('instagram.redirectUri'),
'scope'       => $app['config']->get('instagram.scope'),
```

**Comment by Andreyco** [comment]
@fabnoe please, find updated docs. As pointed by @PanMan, package was on older Laravel 5.2. Now, Larave 5.3+ is supported.
